From 6c08b9c4971f72444b9c862bf253f93e52d6e688 Mon Sep 17 00:00:00 2001
From: Sergey Matyukevich <geomatsi@gmail.com>
Date: Thu, 12 Nov 2020 22:38:09 +0300
Subject: [PATCH 3/3] ARM: dts: imx6sx-udoo-neo: add rproc/rpmsg blocks

Add both remoteproc and rpmsg declarations into Udoo Neo Full DTS tree.
Enable remoteproc. For the time being keep rpmsg disabled to avoid
conflicts between remoteproc and rpmsg. Unit for rpmsg is used to
attach to cortex-m4 firmware that has been started by SCU or U-boot.

Signed-off-by: Sergey Matyukevich <geomatsi@gmail.com>
---
 arch/arm/boot/dts/imx6sx-udoo-neo-full.dts | 40 ++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts b/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
index 96f4d89848a3..84733ce4b3a4 100644
--- a/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
+++ b/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
@@ -15,6 +15,40 @@
 		device_type = "memory";
 		reg = <0x80000000 0x40000000>;
 	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		vdev0vring0: vdev0vring0@bff00000 {
+                        compatible = "shared-dma-pool";
+			reg = <0xbff00000 0x8000>;
+			no-map;
+		};
+		vdev0vring1: vdev0vring1@bff08000 {
+                        compatible = "shared-dma-pool";
+			reg = <0xbff08000 0x8000>;
+			no-map;
+		};
+		vdevbuffer: vdevbuffer {
+                        compatible = "shared-dma-pool";
+			reg = <0xbff10000 0x70000>;
+			no-map;
+		};
+	};
+
+	imx6sx-cm4 {
+		compatible	= "fsl,imx6sx-cm4";
+		syscon		= <&src>;
+		clocks		= <&clks IMX6SX_CLK_M4>;
+		mbox-names = "tx", "rx", "txdb", "rxdb";
+		mboxes = <&mu 0 1
+			  &mu 1 1
+			  &mu 2 1
+			  &mu 3 1>;
+		memory-region = <&vdev0vring0>, <&vdev0vring1>, <&vdevbuffer>;
+	};
 };
 
 &fec1 {
@@ -39,3 +73,9 @@
 &uart3 { /* Bluetooth */
 	status = "okay";
 };
+
+&rpmsg {
+	vdev-nums = <1>;
+	reg = <0xbff00000 0x80000>;
+	status = "disabled";
+};
-- 
2.29.2

